src/app/pages/deposits/deposits.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ToastrService } from 'ngx-toastr';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Subscription } from 'rxjs';
+import { Router } from '@angular/router';
+
+import { Deposit } from 'src/app/models/Deposits';
+import { DepositsService } from 'src/app/services/deposits/deposits.service';
+import { removeSubscription } from 'src/app/helpers/unsubscribe';
+
+@Component({
+  selector: 'app-deposits',
+  templateUrl: './deposits.component.html',
+  styleUrls: ['./deposits.component.scss']
+})
+export class DepositsComponent implements OnInit, OnDestroy {
+  subscribe: Subscription[] = [];
+  headElements = ['Reference', 'Date', 'Amount'];
+  transactions: Deposit[] = [];
+  startDateVal: string = '';
+  endDateVal: string = '';
+  newResults: any[];
+
+  constructor(
+    private depositsService: DepositsService,
+    private spinner: NgxSpinnerService,
+    private toastrService: ToastrService,
+    private router: Router
+  ) {}
+
+  onFilter() {
+    let filteredTransactions = [];
+    this.transactions.filter(transaction => {
+      let date = new Date(transaction['created_at']);
+      let fromDate = new Date(this.startDateVal);
+      let toDate = new Date(this.endDateVal);
+
+      if (fromDate && toDate) {
+        if (date >= fromDate && date <= toDate) {
+          filteredTransactions.push(transaction);
+        }
+      }
+    });
+    this.transactions = filteredTransactions;
+    this.newResults = filteredTransactions;
+  }
+
+  clear() {
+    this.newResults = null;
+    this.setTransactions();
+  }
+
+  ngOnInit() {
+    this.setTransactions();
+  }
+
+  setTransactions() {
+    this.spinner.show();
+    this.subscribe.push(
+      this.depositsService.getDeposits().subscribe(
+        response => {
+          if (response.results.length === 0) {
+            console.log('WHAAAATTT');
+            this.router.navigate(['no-deposits']);
+          } else {
+            console.log('transactions-else ', this.transactions);
+            this.transactions = response.results;
+            this.spinner.hide();
+          }
+        },
+        error => {
+          console.log('Error ', error);
```

Comment:
do you still need this?, if not can we please remove and also the corresponding console.logs

Reply:
Hey @kdaglas , thanks I have removed it